realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -354,6 +357,39 @@ public String getAccessToken() {
         return (userToken != null) ? userToken.value() : null;
     }
 
+    /**
+     * Returns an instance of the Management Realm owned by the user.
+     *
+     * This Realm can be used to control access and permissions for Realms owned by the user. This includes
+     * giving other users access to Realms.
+     *
+     * @see <a href="https://realm.io/docs/realm-object-server/#permissions">How to control permissions</a>
+     */
+    public synchronized Realm getManagementRealm() {
+        if (managementRealmConfig == null) {
+            String managementUrl = getManagementRealmUrl(syncUser.getAuthenticationUrl());
+            managementRealmConfig = new SyncConfiguration.Builder(this, managementUrl)
+                    .modules(new PermissionModule())
+                    .build();
+        }
+
+        return Realm.getInstance(managementRealmConfig);
```

Comment:
We don't need to put `return Realm.getInstance(managementRealmConfig);` inside of the `synchronized`.

It's better to make a `synchronized` block and move `if (managementRealmConfig == null) {...}` into it.